docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,531 @@
+
+#### 4. Commit changes to the local feature branch
+
+When you are pleased with the work you have done on the local copy of your
+feature branch, you will want to `stage` your changes in preparation for a
+`commit`.
+
+To find out which files have been modified, added, deleted, and also what is
+staged or unstaged, run:
+
+```
+git status
+```
+
+To stage all untracked files, run:
+
+```
+git add --all
+```
+
+To stage one untracked file at a time, run:
+
+```
+git add  { relative path to file from root directory }
+```
+
+When you are pleased with your staged changes, it is time to create a `commit`
+and give it a message.
+
+Our commit messages are formatted according to
+[Conventional Commits](https://conventionalcommits.org/).
+
+Please read the
+[Commit Message Format](https://loopback.io/doc/en/contrib/code-contrib-lb4.html#commit-message-guidelines)
+guidelines to correctly format your commit messages.
+
+To help with abiding by the rules of commit messages, please use the
+`commitizen` tool mentioned in the documentation above. This means will we use
+`git cz` instead of `git commit`. Install `commitizen` by running
+`npm install -g commitizen`.
+
+In my case, I want the commit message to look like this:
+
+```
+docs: tutorial on how to submit a pull request to lb4
+
+This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to LoopBack v4
+
+```
+
+To start `commitizen`, run:
+
+```
+git cz
+```
+
+Follow the prompts for the various fields.
+
+In my case I have entered :
+
+```
+type                : docs
+scope               : <blank>
+short description   : tutorial on how to submit a pull request to lb4
+long description    : This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to LoopBack v4
+breaking changes    : N
+affects open issues : N
+```
+
+![submit_pr_git_cz_1.png](./imgs/submit_pr_git_cz_1.png)
+
+The interactive `commitizen` prompts complete and the commit is created with a
+properly formatted message.
+
+#### 5. Push commits to the remote feature branch
+
+It is now time to push your local committed changes of your local feature branch
+directory to your remote feature branch; to keep them in sync.
+
+The very `first` time you push your changes you should run:
+
+```
+git push --set-upstream origin emonddr-doc-changes
+```
+
+The `--set-upstream` parameter will set up the tracking between your local
+feature branch and your remote feature branch.
+
+Afterwards, for any additional push, you just need to run:
+
+```
+git push
+```
+
+It is not necessary to create a pull request immediately on the push of your
+first commit; this can be done later.
+
+Do take some time to think about how many commits you want in the pull request
+when you eventually create it.
+
+If it makes sense to place your changes into one commit, then do so.
+
+If, however, it makes more sense to place your changes into a few logically
+separate commits, then do so.
+
```

Comment:
This paragraph does not add much value unless we have some best practice for them to follow.